### ISSUE TITLE
[Bug]: HeaderMenu component shortens url to `#`

### ISSUE THREAD

**Issue by angella-qian** [carbon-design-system/carbon#10946]
### Question for Carbon

[I've tried posting in our `#carbon-react` channel](https://ibm-analytics.slack.com/archives/C2K6RFJ1G/p1645754997826539) but haven't received any responses there so wanted to post and see if I could get any help with this here.

Wondering if someone has come across this or if this is a known Carbon bug. We’re seeing that anytime you click our `HeaderMenu` component, the url shortens to `#`. We're using the `HeaderMenu` here to allow users to create a new assistant or switch between existing assistants. I see on the implicit `<a>` tag there’s an href set to `#` so I’m thinking that’s what’s causing it. I’m unable to set a `event.preventDefault()` or `event.stopPropagation()` on this `<a>` tag though - I’ve tried adding it to the onClick handler of `HeaderMenuItem` and more. 

![URL](https://user-images.githubusercontent.com/36805013/157506317-70688830-6283-4496-a766-bcc34c60526f.gif)

Related code:
```
  const openCreateModal = useCallback(() => {
    setIsCreateModalOpen(true);
  }, []);

  const handleSwitchAssistant = useCallback(
    id => {
      history.push(
        urlJoin(encodeURIComponent(instance), 'assistants', id, 'home')
      );
    },
    [history, instance]
  );

 <HeaderNavigation>
       <HeaderMenu
            menuLinkName={menuName}>
            <HeaderMenuItem onClick={openCreateModal}>
                <div>
                   Create New
                   <Add24 />
                 </div>
            </HeaderMenuItem>
            {assistants
              .filter(a => a.assistant_id !== assistant?.assistant_id) // Ignore the selected assistant
              .map(a => (
                  <HeaderMenuItem
                     onClick={() => handleSwitchAssistant(a.assistant_id)}
                     key={a.assistant_id}>
                        {a.name ?? 'Unnamed'}
                   </HeaderMenuItem>
              ))}
       </HeaderMenu>
</HeaderNavigation>
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by joshblack** [comment]
Hi there, @angella-qian! 👋  Thanks for taking the time to put this issue together 🙏 

I went ahead and opened up #10967 to remove the `href="#"` that is being set with `HeaderMenu`. For next steps, I'm going to double check with the team that this makes sense from an accessibility perspective. If it all checks out, we can merge it in.

Hope this makes sense! Let me know if you have any questions.

**Comment by angella-qian** [comment]
@joshblack Hey, yep! And thanks for the quick response – just wondering if you have an eta of when this fix will go out?

**Comment by miniwangdali** [comment]
@joshblack Hey, I notice that the fix was closed without merging. Are we still going to have another fix to resolve the problem?

**Comment by jnm2377** [comment]
Hi @angella-qian @miniwangdali there were some a11y concerns with the PR implementation that was opened for this issue. Unfortunately @joshblack has actually moved on to a new team 😢  At the moment, we don't have any further explorations planned, but we would welcome a PR to fix this 🙏🏽

**Comment by guidari** [comment]
Hey @angella-qian 
I was checking this issue and looks like have been fixed. I couldn't reproduce in the [storybook](http://localhost:3000/iframe.html?id=components-ui-shell-header--header-w-navigation&viewMode=story). The "#" is not being placed in the URL.

If the error persists, could you reproduce it on [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon/tree/main/examples/vite?file=src%2FApp.jsx), please?

**Comment by guidari** [comment]
We will close this, but if you find this issue is not actually resolved, please [open a new issue](https://github.com/carbon-design-system/carbon/issues/new/choose) and include a link back to this issue.
